documentation/internal/resource-manager-active-directory.md

```diff
@@ -0,0 +1,292 @@
+
+The last scenario that we would possibly use the generic `HttpClient` for this cmdlet would be if the user provided either the `-ApiVersion` or `-ExpandProperties` parameters. The `-ApiVersion` parameter is used to define which API version of the service should be used when retrieving the resource(s), which can change the values in the `Properties` property on the returned object(s). The `-ExpandProperties` switch parameter is used to signal that the `Properties` property on the returned object(s) should be expanded and include all resource specific information. The `ResourceManagerClient` doesn't allow for either of these scenarios (outside of providing an API version for a resource ID, which was covered previously), so the `HttpClient` must be used to provide users with this functionality.
+
+For all other scenarios, the `ResourceManagerClient` is used either to [list all resources in a given resource group](https://github.com/Azure/azure-powershell/blob/master/src/Resources/ResourceManager/Implementation/Resource/GetAzureResourceCmdlet.cs#L178) or [list all resources in the current subscription](https://github.com/Azure/azure-powershell/blob/master/src/Resources/ResourceManager/Implementation/Resource/GetAzureResourceCmdlet.cs#L182).
+
+**New-AzResource**
+
+Unlike the `Get-AzResource` cmdlet, the `New-AzResource` cmdlet only uses the generic `HttpClient` to make calls to the server to create a resource. With this cmdlet, a user can create a resource at the tenant or subscription level against a specified API version with a defined set of properties. The cmdlet [constructs the resource ID](https://github.com/Azure/azure-powershell/blob/master/src/Resources/ResourceManager/Implementation/Resource/NewAzureResourceCmdlet.cs#L96) from the given identity parameters, [sets up the body](https://github.com/Azure/azure-powershell/blob/master/src/Resources/ResourceManager/Implementation/Resource/NewAzureResourceCmdlet.cs#L105) of the request from the property parameters, and [sends the `PUT` request](https://github.com/Azure/azure-powershell/blob/master/src/Resources/ResourceManager/Implementation/Resource/NewAzureResourceCmdlet.cs#L107-L124) to the server using the `HttpClient` in order to create the resource.
+
+In the future, it should be determined whether or not the `ResourceManagerClient` can be used to make the subscription level calls as it has two functions, `CreateOrUpdate()` and `CreateOrUpdateById()`, which should be able to perform the same steps outlined above.
+
+**Set-AzResource**
```

Comment:
Excuse me, the behavior that a cmdlet could perform 2 different operations, and the operations are differenciated by a switch parameter - is it a common practice? Didn't see that before 🤔

Reply:
@isra-fel this isn't common practice and is advised against in almost every scenario -- we want to follow the pattern `PATCH` --> `Update-*` and `PUT` --> `Set-*`.

This is a very rare case where the cmdlet will handle both operations based on a switch parameter and should not be followed moving forward. This cmdlet was implemented long before we put our current patterns and standards in place, and this cmdlet uses the generic `HttpClient` to make it's updates to the resources, so my guess is that, at the time, it was easier to support both `PUT` and `PATCH` in the cmdlet by just switching what operation was used by the `HttpClient` since it didn't require a separate SDK call.

Let me know if you still have any questions about this.